src/Disks/ObjectStorages/Web/WebObjectStorage.cpp

```diff
@@ -162,7 +162,22 @@ bool WebObjectStorage::exists(const std::string & path) const
 
     if (startsWith(it->first, path)
         || (it != files.begin() && startsWith(std::prev(it)->first, path)))
+    {
+        shared_lock.unlock();
+        std::unique_lock unique_lock(metadata_mutex);
```

Comment:
Why do we need `.unlock()` and `unique_lock` here?

Reply:
I'm not familiar with this code, but looks like yes. `exists` is such a function which on a good-path is read-only (need only shared_lock`), but on a bad path can modify state itself. So here we modify state similar to https://github.com/ClickHouse/ClickHouse/pull/55468/files#diff-64938a2d590929ddef45935da30a0458bcf7181c6585f34fdf7cebff96be3990R132-R144. That is why we releasing shared lock and acquiring write lock (unique_lock).